ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx

```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
+    await pickMainCpu(container, 8);
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(MIN_DISK_SIZE_GB + 10);
+
+    // GPU
+    await pickPresets(container, runtimePresets.generalAnalysis.displayName);
+
+    clickExpectedButton('Create');
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual('Creating');
+      expect(runtimeApiStub.runtime.configurationType).toEqual(
+        RuntimeConfigurationType.GENERAL_ANALYSIS
+      );
+      expect(runtimeApiStub.runtime.gceWithPdConfig.persistentDisk).toEqual({
+        diskType: 'pd-standard',
+        labels: {},
+        name: 'stub-disk',
+        size: MIN_DISK_SIZE_GB,
+      });
+      expect(runtimeApiStub.runtime.dataprocConfig).toBeFalsy();
+    });
+  });
+
+  it('should allow configuration via dataproc preset', async () => {
+    setCurrentRuntime(null);
+
+    const { container } = component();
+
+    const CustomizeButton = screen.getByRole('button', {
+      name: 'Customize',
+    });
+
+    await CustomizeButton.click();
+
+    await pickPresets(container, runtimePresets.hailAnalysis.displayName);
+
+    clickExpectedButton('Create');
+    await waitFor(
+      () => {
+        expect(runtimeApiStub.runtime.status).toEqual('Creating');
+        expect(runtimeApiStub.runtime.configurationType).toEqual(
+          RuntimeConfigurationType.HAIL_GENOMIC_ANALYSIS
+        );
+        expect(runtimeApiStub.runtime.dataprocConfig).toEqual(
+          runtimePresets.hailAnalysis.runtimeTemplate.dataprocConfig
+        );
+        expect(runtimeApiStub.runtime.gceConfig).toBeFalsy();
+      },
+      { interval: 750 }
+    );
+  });
+
+  it(
+    'should set runtime preset values in customize panel instead of getRuntime values ' +
+      'if configurationType is GeneralAnalysis',
+    async () => {
+      const customMachineType = 'n1-standard-16';
+      const customDiskSize = 1000;
+      setCurrentRuntime({
+        ...runtimeApiStub.runtime,
+        status: RuntimeStatus.DELETED,
+        configurationType: RuntimeConfigurationType.GENERAL_ANALYSIS,
+        gceConfig: {
+          ...defaultGceConfig(),
+          machineType: customMachineType,
+          diskSize: customDiskSize,
+        },
+        dataprocConfig: null,
+      });
+
+      // show that the preset values do not match the existing runtime
+
+      const { machineType, persistentDisk } =
+        runtimePresets.generalAnalysis.runtimeTemplate.gceWithPdConfig;
+
+      expect(customMachineType).not.toEqual(machineType);
+      expect(customDiskSize).not.toEqual(persistentDisk.size);
+      const { container } = component();
+      // Try :Check there is no spinner
+      expect(getMainCpu(container)).toEqual(
+        findMachineByName(machineType).cpu.toString()
+      );
+      expect(getMainRam(container)).toEqual(
+        findMachineByName(machineType).memory.toString()
+      );
+      expect(getDetachableDiskValue()).toEqual('120');
```

Comment:
it's not obvious what 120 represents here, so it should be `persistentDisk.size`

Reply:
Updated